src/components/hud/ActiveProjectDisplay.tsx

```diff
@@ -1,86 +1,72 @@
+  const bufferProgress = progress + value / props.projectDefinition.effort * 100
+  return (
+    <Grid container spacing={3}>
+      <Grid item xs={12}><LinearProgress variant='buffer' value={progress} valueBuffer={bufferProgress} /></Grid>
+      <Grid item xs={9}><Typography>{props.projectDefinition.description}</Typography></Grid>
+      <Grid item xs={3} sx={{ justifySelf: 'end' }}>
+        👤<Input
+          value={value}
+          size='small'
+          onChange={props.onWorkersChange}
+          inputProps={{
+            step: 1,
+            min,
+            max,
+            type: 'number',
+            pattern: '[0-9]+'
+          }}
+          />
+      </Grid>
     </Grid>
   )
+}
 
-  const grid1 = (
-    <Grid item xs>
-      <Slider
-        value={typeof value === 'number' ? value : 0}
-        min={min}
-        max={max}
-        onChange={onWorkersChange}
-        aria-labelledby='input-slider'
-      />
-    </Grid>
-  )
+export function ActiveProjectDisplay (props: {activeProject: ActiveProject, tileIndex: number, workers: WorkerState, dispatch: dispatcher}) {
+  const projectDefinition = projectCatalog[props.activeProject.type]
+  const remainingEffort = projectDefinition.effort - props.activeProject.progress
+  const max = Math.min(props.activeProject.assignedWorkers + props.workers.free, remainingEffort)
+  const onWorkersChange = (event: MouseEvent) => {
+    if (event.target && event.target instanceof HTMLInputElement) {
+      let workerCount = parseInt(event.target.value)
+      if (!isNaN(workerCount)) {
+        workerCount = Math.min(workerCount, max)
+        props.dispatch({ type: 'assignWorkers', tileIndex: props.tileIndex, workerCount })
+      }
+    }
+  }
+
+  const onCancelProject = () => {
+    props.dispatch({ type: 'cancelProject', tileIndex: props.tileIndex })
+  }
 
   return (
-    <div>
-      <Grid container spacing={2}>
-        <Grid item xs={12}>
-          <Typography>
-            Project: {projectDefinition.name}
-          </Typography>
-          <LinearProgress variant='buffer' value={progress} valueBuffer={bufferProgress} />
-        </Grid>
-        <Grid container item xs={6}>
-          <Card>
-            <CardContent>
-              <Typography>
-                {projectDefinition.description}
-              </Typography>
-            </CardContent>
-          </Card>
-        </Grid>
-        <Grid container item xs={6} spacing={2}>
-          <Grid item xs>
-            <Typography gutterBottom>
-              Workers
-            </Typography>
-          </Grid>
-          {grid1}
-          {grid2}
-        </Grid>
-      </Grid>
-    </div>
+    <ProjectCard
+      project={projectDefinition} action={
+        <Button onClick={onCancelProject}>Cancel</Button>
+    }
+      content={
+        activeProjectContent({ activeProject: props.activeProject, projectDefinition: projectDefinition, onWorkersChange: onWorkersChange, workers: props.workers })
```

Comment:
Do not call react components as regular functions! Because if activeProjectContent ever got any hooks, they would instead be appropriated by this component. Getting errors with the angle bracket syntax? React components should be *Upper*CamelCase.